test/unit/pool/test-destroy-connection.js

```diff
@@ -11,10 +11,10 @@ server.listen(common.fakeServerPort, function (err) {
   pool.getConnection(function (err, connection) {
     assert.ifError(err);
 
-    assert.strictEqual(connection, pool._allConnections[0]);
+    assert.strictEqual(connection, pool._connectionManager._allConnections._map[1]);
```

Comment:
I really understand. but, I think it's uncommon to use those properties such as `Pool._acquiringConnections` directly. It seems that it's not easy and efficient to work with all the existing ones.

I thought shortly.
```js
pool.createPool({
    ...,
    keepOldProperties : true
});
```
If the user set this, 4 properties(_acquiringConnections, _allConnections, _freeConnections, _connectionQueue) are updated. Isn't it enough?